components/automate-ui/src/app/entities/notification_rules/notification_rule.reducer.ts

```diff
@@ -33,43 +41,113 @@ export function notificationRuleEntityReducer(
+      return set(
+        GET_ALL_STATUS,
+        EntityStatus.loading,
+        notificationRuleEntityAdapter.removeAll(state)
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_ALL_SUCCESS:
       return pipe(
-        set(GET_ALL_STATUS, EntityStatus.loadingSuccess))
-        (notificationRuleEntityAdapter.setAll(action.payload, state));
+        set(GET_ALL_STATUS,
+          EntityStatus.loadingSuccess)
+        )
+        (notificationRuleEntityAdapter.setAll(action.payload, state)
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_ALL_FAILURE:
-      return set(GET_ALL_STATUS, EntityStatus.loadingFailure, state);
+      return set(
+        GET_ALL_STATUS,
+        EntityStatus.loadingFailure,
+        state
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET:
-      return set(GET_STATUS, EntityStatus.loading,
+      return set(
+        GET_STATUS,
+        EntityStatus.loading,
         notificationRuleEntityAdapter.removeAll(state)
       ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_SUCCESS:
-      return set(GET_STATUS, EntityStatus.loadingSuccess,
+      return set(
+        GET_STATUS,
+        EntityStatus.loadingSuccess,
         notificationRuleEntityAdapter.addOne(action.payload, state)
       ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.GET_FAILURE:
-      return set( GET_STATUS, EntityStatus.loadingFailure,
-        state ) as NotificationRuleEntityState;
+      return set(
+        GET_STATUS,
+        EntityStatus.loadingFailure,
+        state
+      ) as NotificationRuleEntityState;
+
+    case NotificationRuleActionTypes.CREATE: {
+      return set(
+        SAVE_STATUS,
+        EntityStatus.loading,
+        state
+      ) as NotificationRuleEntityState;
+    }
+
+    case NotificationRuleActionTypes.CREATE_SUCCESS: {
+      return pipe(
+          unset(SAVE_ERROR),
+          set(SAVE_STATUS, EntityStatus.loadingSuccess)
+        )(notificationRuleEntityAdapter.addOne(action.payload, state)
+      ) as NotificationRuleEntityState;
+    }
+
+    case NotificationRuleActionTypes.CREATE_FAILURE: {
+      return pipe(
+        set(SAVE_ERROR, action.payload),
+        set(SAVE_STATUS, EntityStatus.loadingFailure)
+      )(state) as NotificationRuleEntityState;
+    }
 
     case NotificationRuleActionTypes.UPDATE:
-      return set(UPDATE_STATUS, EntityStatus.loading, state);
+      return set(
+        UPDATE_STATUS,
+        EntityStatus.loading,
+        state
+      ) as NotificationRuleEntityState;
 
     case NotificationRuleActionTypes.UPDATE_SUCCESS:
       return set(UPDATE_STATUS, EntityStatus.loadingSuccess,
         notificationRuleEntityAdapter.updateOne({
           id: action.payload.id,
           changes: action.payload
-        }, state));
+        }, state)
+      ) as NotificationRuleEntityState;
```

Comment:
Looks like you added the typecast here, but prefer to remove the unneeded ones, not add more.